dev-docs/source/MVPDesign.rst

```diff
@@ -179,6 +179,68 @@ in C++, or `unittest.mock
 can set expectations in the unit tests for certain methods to be
 called, and with certain arguments.
 
+MVP Template tool
+#################
+
+The `template.py script <https://github.com/mantidproject/mantid/blob/main/tools/MVP/template.py>`__
+provides a tool which is designed to generate initial files for an MVP-based
+widget written in either Python or C++. These generated files serve as a
+foundation or template for creating any widget using the MVP design pattern.
+It can also be used to create the necessary files when refactoring an existing
+widget which is not currently using MVP.
+
+Python
+------
+
+To generate files for a Python widget with name "Example", run:
+
+.. code-block:: sh
+
+  python tools/MVP/template.py --name Example --language python --output-dir $PWD/..
+
+
+This command will generate four python files including `example_model.py`, `example_view.py`
+and `example_presenter.py`. These files will be saved in the provided output directory,
+as specified by `$PWD/..`. An additional file named `launch.py` will be generated. This
+can be used to open the widget as follows:
+
+.. code-block:: sh
+
+  python $PWD/../launch.py
+
+C++
+---
+
+To generate files for a C++ widget with name "Example", run:
+
+.. code-block:: sh
+
+  python tools/MVP/template.py --name Example --language c++ --output-dir $PWD/..
+
+This command will generate eight files including `ExampleModel.cpp`, `ExampleModel.h`,
+`ExampleView.cpp`, `ExampleView.h`, `ExamplePresenter.cpp` and `ExamplePresenter.h`.
+An additional file named `main.cpp` and a `CMakeLists.txt` will be generated. These
+files can be used to build the widget as follows:
+
+.. code-block:: sh
+
+  mkdir buildmvp
+  cd buildmvp
+  cmake ..
+  cmake --build .
+
+The example widget can then be opened with:
+
+.. code-block:: sh
+
+  cd buildmvp
+  ./Debug/launch.exe
```

Comment:
On windows it generates by default with visual studio. But in MacOs (and I guess Linux too) it uses Makefiles. 
In this case, the output does not have .exe extension and is not in debug folder. So could you add a small line indicating that in Unix systems the launch instructions for cpp should be:
```suggestion
  ./launch
```

Also the `./Debug/launch.exe` won't work in windows unless called from a shell or bash. You can tip to use the GitBash to run this example on windows.